plugins/widget/plugin.js

```diff
@@ -2901,6 +3004,80 @@
 		} );
 	}
 
+	function handleStylesOnDowncast( widget, widgetElement ) {
+		var styledElement;
+
+		if ( widgetElement ) {
+			if ( widget.styleDefinition && widgetElement.name !== widget.styleDefinition.element ) {
+				styledElement = widgetElement.getFirst( widget.styleDefinition.element );
+			}
+			if ( !styledElement ) {
+				styledElement = widgetElement;
+			}
+
+			styledElement.attributes = styledElement.attributes || {};
+			if ( widget.styleDefinition && widget.styleDefinition.styles ) {
+				styledElement.attributes.style = CKEDITOR.tools.writeCssText( widget.styleDefinition.styles );
+			}
+		}
+	}
+
+	function handleStylesOnUpcast( element, wrapper ) {
+		var style = element.attributes.style,
+			lockedStyle = {},
+			outputStyle = {},
+			styleDefinition = {},
+			key;
+
+		if ( element.attributes[ 'data-cke-widget-data' ] ) {
+			lockedStyle = JSON.parse( decodeURIComponent( element.attributes[ 'data-cke-widget-data' ] ) ).lockedStyle || {};
```

Comment:
Probably it would be safer to delete `lockedStyle` property from widget's data after handling styles as it could pollute it. Maybe it would be even better to move locked styles to dedicated `[data-cke-lockedstyle]` property? 🤔

Reply:
As we discussed I'm moving `lockedStyle` to `styleDefinition.lockedStyle`.